### PR TITLE
[SB] Batch size update

### PR DESCRIPTION
**Pull request by l0lawrence**

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py [hunk 1]
```diff
@@ -135,6 +135,8 @@
 DEAD_LETTER_SUPPLEMENTARY_AUTHORIZATION_HEADER = "ServiceBusDlqSupplementaryAuthorization"
 
 MAX_MESSAGE_LENGTH_BYTES = 1024 * 1024  # Backcompat with uAMQP
+MAX_BATCH_SIZE_STANDARD = 256000
+MAX_BATCH_SIZE_PREMIUM = 1000000
```
Comment (l0lawrence) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py:139]: 1024 * 1024

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py [hunk 2]
```diff
@@ -135,6 +135,8 @@
 DEAD_LETTER_SUPPLEMENTARY_AUTHORIZATION_HEADER = "ServiceBusDlqSupplementaryAuthorization"
 
 MAX_MESSAGE_LENGTH_BYTES = 1024 * 1024  # Backcompat with uAMQP
+MAX_BATCH_SIZE_STANDARD = 256000
```
Comment (l0lawrence) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py:138]: 1024 * 256

Comment (l0lawrence) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py:138]:
```suggestion
MAX_BATCH_SIZE_STANDARD = 256 * 1024
```

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py [hunk 3]
```diff
@@ -135,6 +135,8 @@
 DEAD_LETTER_SUPPLEMENTARY_AUTHORIZATION_HEADER = "ServiceBusDlqSupplementaryAuthorization"
 
 MAX_MESSAGE_LENGTH_BYTES = 1024 * 1024  # Backcompat with uAMQP
+MAX_BATCH_SIZE_STANDARD = 256 * 1024
+MAX_BATCH_SIZE_PREMIUM = 1000000
```
Comment (l0lawrence) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py:139]:
```suggestion
MAX_BATCH_SIZE_PREMIUM = 1024 * 1024
```

File: sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py [hunk 4]
```diff
@@ -1996,6 +1996,27 @@ def message_content():
...
+    ):
+
+        fully_qualified_namespace = f"{servicebus_namespace.name}{SERVICEBUS_ENDPOINT_SUFFIX}"
+        credential = get_credential(is_async=True)
+        async with ServiceBusClient(
+            fully_qualified_namespace, credential, logging_enable=False, uamqp_transport=uamqp_transport
+        ) as sb_client:
+            async with sb_client.get_queue_sender(servicebus_queue.name) as sender:
+                with pytest.raises(ValueError):
+                    batch_message = [ServiceBusMessage("test message " * 256 * 1024)]
```
Comment (l0lawrence) [inline sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py:2017]: larger than 256 kb

Comment (kashifkhan) [reply sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py:2017]: This test will fail in order to pass ?

we should add in tests for add as well.

Comment (l0lawrence) [reply sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py:2017]: we have tests for batch for passing already

Comment (kashifkhan) [reply sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py:2017]: there are ones that check for `MessageSizeExceededError` ? if so then cool :)

Comment (l0lawrence) [reply sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py:2017]: https://github.com/Azure/azure-sdk-for-python/blob/480ed8878e88aba478ed6132def8d51bdf0cc909/sdk/servicebus/azure-servicebus/tests/test_message.py#L216 they catch ValueError which MessageSizeExceededError extends from

File: sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_sender_async.py [hunk 5]
```diff
@@ -458,14 +460,19 @@ async def create_message_batch(self, max_size_in_bytes: Optional[int] = None) ->
         if not self._max_message_size_on_link:
             await self._open_with_retry()
 
-        if max_size_in_bytes and max_size_in_bytes > self._max_message_size_on_link:
+        if self._max_message_size_on_link > MAX_BATCH_SIZE_PREMIUM:
+            max_size_allowed = MAX_BATCH_SIZE_PREMIUM
```
Comment (kashifkhan) [inline sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_sender_async.py:464]: lets make `max_size_allowed` be something like  `self._max_batch_size_on_link` and populated right after here https://github.com/Azure/azure-sdk-for-python/blob/480ed8878e88aba478ed6132def8d51bdf0cc909/sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py#L549 . This way if the metadata from the service ever returns the right size itll be a cleaner update.

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_sender.py [hunk 6]
```diff
@@ -82,6 +84,7 @@ def _create_attribute(self, **kwargs):
         self._error_policy = self._amqp_transport.create_retry_policy(self._config)
         self._name = kwargs.get("client_identifier") or f"SBSender-{uuid.uuid4()}"
         self._max_message_size_on_link = 0
```
Comment (l0lawrence) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_sender.py:86]: why do we set these both twice?

Comment (kashifkhan) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_sender.py:86]: is `create_attribute` called elsewhere ? otw yeah might be a mistake

**Review by kashifkhan — APPROVED** [review]

**Pull request merged**